socketio/src/asynchronous/client/client.rs

```diff
@@ -87,11 +88,25 @@ impl Client {
             nsp: builder.namespace.to_owned(),
             outstanding_acks: Arc::new(RwLock::new(Vec::new())),
             auth: builder.auth.clone(),
+            data: builder.data.clone().unwrap_or(Arc::new(())),
             builder: Arc::new(RwLock::new(builder)),
             disconnect_reason: Arc::new(RwLock::new(DisconnectReason::default())),
         })
     }
 
+    /// Fetches data given by [`ClientBuilder::data`]
+    pub fn data<D: Send + Sync + 'static>(&self) -> Arc<D> {
+        self.try_data()
+            .expect("Client::data does not match ClientBuilder::data")
```

Comment:
Mhmmm... I don't really like that we need a runtime check here... Wdyt, would it be possible to make this a generic parameter of both the Client and the builder? If this turns out messy, feel free to omit.

Reply:
Generic parameter was my main concern, sounds like it'd get messy and fast.